asana/asana.go

```diff
@@ -189,8 +183,39 @@ type (
 
 	// Errors always has at least 1 element when returned.
 	Errors []Error
+
+	RequestError struct {
+		Err      error
+		Response *http.Response
+		Text     string
+	}
 )
 
+func (re RequestError) New(resp *http.Response, err error) RequestError {
+	txt := "asana: error"
+	switch resp.StatusCode {
+	case http.StatusBadRequest:
+		txt = TextBadRequest
+	case http.StatusUnauthorized:
+		txt = TextUnauthorized
+	case http.StatusPaymentRequired:
+		txt = TextPaymentRequired
+	case http.StatusForbidden:
+		txt = TextForbidden
+	case http.StatusNotFound:
+		txt = TextNotFound
+	case http.StatusTooManyRequests:
+		txt = TextThrottled
+	case http.StatusInternalServerError:
+		txt = TextInternal
+	}
+	return RequestError{Response: resp, Err: err, Text: txt}
+}
+
+func (re RequestError) Error() string {
```

Comment:
I believe this is satisfying the `error` interface, but unsure if this is bad for our logs. Do logs use this? Or do they just include the whole error object?

Reply:
This is necessary and seems fine.